cli/flox-rust-sdk/src/models/environment/path_environment.rs

```diff
@@ -555,23 +594,58 @@ pub mod test_helpers {
         )
     }
 
+    pub fn new_named_path_environment(flox: &Flox, contents: &str, name: &str) -> PathEnvironment {
+        new_named_path_environment_in(
+            flox,
+            contents,
+            tempdir_in(&flox.temp_dir).unwrap().into_path(),
+            name,
+        )
+    }
+
     pub fn new_path_environment_from_env_files(
         flox: &Flox,
         env_files_dir: impl AsRef<Path>,
     ) -> PathEnvironment {
         let dot_flox_parent_path = tempdir_in(&flox.temp_dir).unwrap().into_path();
-        new_path_environment_from_env_files_in(flox, env_files_dir, dot_flox_parent_path)
+        new_path_environment_from_env_files_in(flox, env_files_dir, dot_flox_parent_path, None)
+    }
+
+    pub fn new_named_path_environment_from_env_files(
+        flox: &Flox,
+        env_files_dir: impl AsRef<Path>,
+        name: &str,
+    ) -> PathEnvironment {
+        let dot_flox_parent_path = tempdir_in(&flox.temp_dir).unwrap().into_path();
+        new_path_environment_from_env_files_in(
+            flox,
+            env_files_dir,
+            dot_flox_parent_path,
+            Some(name),
+        )
```

Comment:
Having so many helpers is getting pretty gross. We should clean it up in https://github.com/flox/flox/issues/1682

Reply:
(non-blocking) could also just use a randomly generated name (e.g. the tempdir name you fall back to) and introspect it from `env.name()`.

Reply:
Yeah that would be nicer in some ways. Another part of me prefers how some tests currently assert against output containing the name without constructing it programmatically. I don't feel super strongly but don't think I'll change it at the moment